### ISSUE TITLE
Research Paper Chatbot

### ISSUE THREAD

**Issue by Jamesduongrx** [mikeizbicki/cmc-csci181-languages#42]
@EthanTu2

[CS-181 Final Project.pdf](https://github.com/user-attachments/files/17628853/CS-181.Final.Project.pdf)

**Comment by mikeizbicki** [comment]
This rubric needs significant revision because your grading criteria are not specific enough.  For example:

Chatbot efficiency: This is a marketing description of what your chatbot will do.  You say that you will be able to "ensure accurate queries"  without saying what those queries are or what techniques you will use.  If you are not able to answer queries to my satisfaction (which you probably won't be able to since I have a habit of designing particularly evil queries), then you will get a 0 for this section as written.  So don't be a marketer.  Be an engineer.  Be specific about exactly what you will do so that I cannot misunderstand you.  Are you using RAG?  How will you decide what the documents are that are being searched?  How will you deal with figures?  How will you deal with equations?  What domain of documents will you be using?  What is your source for these documents?  These are all extremely important questions that you don't address at all.

Github cohesiveness: What are the artifacts?  Is it code?  Is it a SQL database?  Is it the papers themselves?

Error Handling: You say "implement and document comprehensive error-handling protocols".  I have no idea what this means.

Testing and Validation: I don't know what it means to test "accuracy against ChatGPT".  What dataset will you use?  A standard one or one that you will create?  If it's one that you will create, how will you create it?

Also, remove the presentation and reflection sections.

**Comment by EthanTu2** [comment]
[CS-181 Final Project.pdf](https://github.com/user-attachments/files/17812237/CS-181.Final.Project.pdf)

Updated Rubric!

**Comment by mikeizbicki** [comment]
Here are my initial thoughts on your rubric.  I would be happy to chat in more detail Friday.

"Using Large Language Models" section: all these tasks are reasonable.

"Implementing RAG" section:

- Basic questions: neither of your two examples are related to RAG (at least how we talked about it in class).  The question "what is Apple Inc's latest stock price" would best be answered by calling out to an API instead of loading a text document.  (See <https://console.groq.com/docs/tool-use> for examples of how to do this.)  The question "What sector does Apple operate in" doesn't need RAG to answer because the model already knows the answer without using loading in additional info into the context.  My guess is that every public company is well represented in the training data.  (Certainly all of their legally required SEC submission documents will be trained on.)  RAG would only (possibly) be needed to answer information about non-public companies, which I believe is outside the scope of your project.
- The advanced questions are better.  But your first example "is Apple a good buy" is very generic.  I don't see how that could meaningfully be tested.  The second example "How might Apple be impacted by recent news on global supply chains?" is better.
- No where do you describe what datasource you will be using for RAG.  This makes it difficult for me to assess whether your goals are reasonable.

Database:
- You should not store stock data in your database.  This will always be out of date.  You should fetch this directly from an API.
- I don't see how market trends can fit into the schema.
- I don't see how API integration is related to a database.
- I suspect your definition of "efficient storage and retrieval" is very different from mine, and that you will not be able to achieve my definition.  In general, you have a lot of ambiguous language like this in your rubric which is dangerous for you.

Testing and Validation:

- You don't mention what data you will be using.  For example, what expert decisions will you be using?
- I don't understand your stress testing section.

**Comment by EthanTu2** [comment]
[CS-181 Final Project.pdf](https://github.com/user-attachments/files/17874608/CS-181.Final.Project.pdf)

Approved Final Project Rubric!
